line_profiler/_line_profiler.pyx

```diff
@@ -1,12 +1,50 @@
-from .python25 cimport PyFrameObject, PyObject, PyStringObject
+from python25 cimport PyFrameObject, PyObject, PyStringObject
+from sys import byteorder
+cimport cython
+from cpython.version cimport PY_VERSION_HEX
+from libc.stdint cimport int64_t
 
+from libcpp.unordered_map cimport unordered_map
+
+# long long int is at least 64 bytes assuming c99
+ctypedef unsigned long long int uint64
+ctypedef long long int int64
 
 cdef extern from "frameobject.h":
+    """
+    inline PyObject* get_frame_code(PyFrameObject* frame) {
+        #if PY_VERSION_HEX < 0x030B0000
+            Py_INCREF(frame->f_code->co_code);
+            return frame->f_code->co_code;
+        #else
+            PyCodeObject* code = PyFrame_GetCode(frame);
+            PyObject* ret = PyCode_GetCode(code);
+            Py_DECREF(code);
+            return ret;
+        #endif
+    }
+    """
+    cdef object get_frame_code(PyFrameObject* frame)
     ctypedef int (*Py_tracefunc)(object self, PyFrameObject *py_frame, int what, PyObject *arg)
 
 cdef extern from "Python.h":
+    """
+    
+    #if PY_VERSION_HEX >= 0x030b00a6
```

Comment:
This is a 3.11 fix due to moving the location of `PyFrameObject`? If so can we add a comment noting that?

Reply:
Yes, sure!